packages/admin-ui/ui/src/components/molecules/item-search/index.tsx

```diff
@@ -0,0 +1,123 @@
+import { AdminGetInventoryItemsParams, ProductVariant } from "@medusajs/medusa"
+import { ControlProps, OptionProps, SingleValue } from "react-select"
+import { InventoryItemDTO, InventoryLevelDTO } from "@medusajs/types"
+
+import Control from "../select/next-select/components/control"
+import { NextSelect } from "../select/next-select"
+import React from "react"
+// TODO: Fix this
+import SearchIcon from "../../fundamentals/icons/search-icon"
+import { useAdminInventoryItems } from "medusa-react"
+import { useDebounce } from "../../../hooks/use-debounce"
+import { useState } from "react"
+
+type Props = {
+  onItemSelect: (item: itemType) => void
+  clearOnSelect?: boolean
+  filters?: AdminGetInventoryItemsParams
+}
+
+type itemType = Partial<InventoryItemDTO> & {
+  location_levels?: InventoryLevelDTO[] | undefined
+  variants?: ProductVariant[] | undefined
+}
+
+type ItemOption = {
+  label: string | undefined
+  value: string | undefined
+  inventoryItem: itemType
+}
+
+const ItemSearch = ({ onItemSelect, clearOnSelect, filters = {} }: Props) => {
+  const [itemSearchTerm, setItemSearchTerm] = useState<string | undefined>()
+
+  const debouncedItemSearchTerm = useDebounce(itemSearchTerm, 500)
+
+  const queryEnabled = !!debouncedItemSearchTerm?.length
+
+  const { isLoading, inventory_items } = useAdminInventoryItems(
+    {
+      q: debouncedItemSearchTerm,
+      ...filters,
+    },
+    { enabled: queryEnabled }
+  )
+
+  const onChange = (item: SingleValue<ItemOption>) => {
+    if (item) {
+      onItemSelect(item.inventoryItem)
+    }
+  }
+
+  const options = inventory_items?.map((inventoryItem) => ({
+    label:
```

Comment:
thought(non-blocking): Given that we can only search for `sku`, it could take highest priority wrt labeling

Reply:
thought(non-blocking): We could also include `inventory_item.title` in the text search on the backend to increase the probability of search results.

Reply:
yeah, going with the second option here, I think we should keep it as is but add the text search + create a migration script 😄 we don't have any way to change titles etc. at the moment though.

Should we add an initial modal for the inventory table to change the title of an item along with it's description? (possibly also thumbnail)

cc: @srindom

Reply:
> Should we add an initial modal for the inventory table to change the title of an item along with it's description?

Sounds good to me. Description and title are fine to start with imo :)